src/main/java/com/internet/shop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -148,7 +150,9 @@ private User getUser(ResultSet resultSet) throws SQLException {
         String name = resultSet.getString("name");
         String login = resultSet.getString("login");
         String password = resultSet.getString("password");
+        byte[] salt = resultSet.getBytes("salt");
         User user = new User(name, login, password);
+        user.setSalt(salt);
         user.setId(id);
         user.setRoles(getRoles(id));
```

Comment:
let's not set roles in this method
this creates a connection inside the connection